src/dataflow/utilities/node.ts

```diff
@@ -117,6 +117,29 @@ export const NodeSensorTypes = [
   },
 ];
 
+export const NodeGeneratorTypes = [
+  {
+    name: "sine",
+    method: (t: number, p: number, a: number, v: number) => Math.round(Math.sin(t * Math.PI / p) * a * 100) / 100
```

Comment:
@mklewandowski I think `p` is being used as wavelength in each of these functions (sine, square and triangle). Changing the frequency in the generator block works in the opposite direction than expected. The quickest fix would be to pass `1/frequency` in `updateGeneratorNode`.

Reply:
Good catch.  I believe that we actually want the blocks to behave as they do currently but change the label on the block to use `period` instead of `frequency`.